client/src/margo_client.c

```diff
@@ -437,6 +448,76 @@ int invoke_client_truncate_rpc(int gfid, size_t filesize)
     return (int)ret;
 }
 
+/* invokes the client unlink rpc function */
+int invoke_client_unlink_rpc(int gfid)
+{
+    /* check that we have initialized margo */
+    if (NULL == client_rpc_context) {
+        return UNIFYFS_FAILURE;
+    }
+
+    /* get handle to rpc function */
+    hg_handle_t handle = create_handle(client_rpc_context->rpcs.unlink_id);
+
+    /* fill in input struct */
+    unifyfs_unlink_in_t in;
+    in.app_id         = (int32_t)app_id;
+    in.local_rank_idx = (int32_t)local_rank_idx;
+    in.gfid           = (int32_t)gfid;
+
+    /* call rpc function */
+    LOGDBG("invoking the unlink rpc function in client");
+    hg_return_t hret = margo_forward(handle, &in);
+    assert(hret == HG_SUCCESS);
+
+    /* decode response */
+    unifyfs_filesize_out_t out;
+    hret = margo_get_output(handle, &out);
+    assert(hret == HG_SUCCESS);
+    int32_t ret = out.ret;
+    LOGDBG("Got response ret=%" PRIi32, ret);
+
+    /* free resources */
+    margo_free_output(handle, &out);
+    margo_destroy(handle);
+    return (int)ret;
+}
+
+/* invokes the client-to-server laminate rpc function */
```

Comment:
Add to this comment: "We assume the gfid is not laminated."  ..to make it clear that the developer should first check if the file is laminated before doing the RPC (we currently do this check in chmod()).

Reply:
I am thinking it'd be fine to have the laminate rpc act as a NOP in the case the file is already laminated, so it's fine to call the laminate_rpc multiple times on a file.  We could return an error like ELAMINATED in that case if the user wanted to know.

Reply:
So are you saying you'd add a check in `invoke_client_laminate_rpc()` to see if the locally cached `is_laminated` is already set, and just return immediately?  Or are you saying doing a laminate RPM on an already laminated file is harmless and essentially a NOP?

Reply:
Right now the server-side rpc just happily runs through all of the laminate logic a second time.  That won't hurt.  However, we could probably tweak this to check first, skip it if already laminated, and return an ELAMINATED error.  Can look at that later.

Reply:
Ok, thanks

Reply:
For now, I'll just leave this as the "run though again, but it doesn't hurt to re-laminate a laminated file" implementation.